src/mod/applications/mod_valet_parking/mod_valet_parking.c

```diff
@@ -86,6 +86,36 @@ static switch_status_t valet_on_dtmf(switch_core_session_t *session, void *input
 	case SWITCH_INPUT_TYPE_DTMF:
 		{
 			switch_dtmf_t *dtmf = (switch_dtmf_t *) input;
+			switch_channel_t *channel = switch_core_session_get_channel(session);
+			const char *terminators;
+
+			if ((terminators = switch_channel_get_variable(channel, "valet_parking_orbit_terminators"))) {
```

Comment:
I don't like doing get_variable in every itteration where a user hits a dtmf due to it allocating memory each time you call that.  We should instead do:

switch_channel_set_private(channel, "orbit_terminators", switch_channel_get_variable(channel, "valet_parking_orbit_terminators"));

when we set up the parking and use switch_channel_get_private to get it in this block of code.  we could also pre-validate the other info below and skip all this if there is nowhere to transfer to.  if doing all this before, you can make a struct allocated on session pool to store all the settings or could convert _orbit_exit_key_ to a struct.